frontend/src/components/grantaccess/GrantAccess.vue

```diff
@@ -234,94 +235,103 @@ function toRequestPayload(
     return request;
 }
 
+const composeAndPushNotificationMessage = (
+    severity: Severity,
+    clientIDList: string[]
+) => {
+    let username = formData.value.userId.toUpperCase();
+    if (severity === Severity.success) {
+        pushNotification(
+            severity,
+            `${username} ${
+                clientIDList[0] === ''
+                    ? `was successfully added with the role ${
+                          getSelectedRole()?.role_name
+                      }`
+                    : `was successfully added with Client IDs: ${clientIDList.join(
+                          ', '
+                      )}`
+            }`
+        );
+    }
+    if (severity === Severity.warning) {
+        pushNotification(
+            severity,
+            `${username} ${
+                clientIDList[0] === ''
+                    ? `already exists with the role ${
+                          getSelectedRole()?.role_name
+                      }`
+                    : `already exists with Client IDs: ${clientIDList.join(
+                          ', '
+                      )}`
+            }`
+        );
+    }
```

Comment:
This is just a question.
The function `composeAndPushNotificationMessage` seems to work.
But looking at this warning if condition, how does the code has the knowledge to know that when it is a "**warning**" then the logic inside this "if" block need to compose a message like these (`role already exists` or `client ID already exists`)? What is the warning (which is **409**) is for something else?

Reply:
We could evaluate the `role already exists` in another way. You are right about other 409 cases. I'll have a look on that.

Reply:
This is just a question. No need to do anything for now. I feel maybe currently probably you won't have a way...
Hi @MCatherine1994 , I am thinking maybe we will need previous ticket discussion on returning some `code` from the backend to decide which waring/error is what for frontend... or good strategy for identifying different errors/warnings.

Reply:
I'll mark this as resolved but we could discuss it and cut a new ticket for that.